src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -652,27 +646,30 @@ private static string GetBracePairString(SyntaxFormattingOptions formattingOptio
 
         private void InsertBraceAndMoveCaret(
             ITextView textView,
-            Document document,
+            ITextBuffer buffer,
+            ParsedDocument document,
             SyntaxFormattingOptions formattingOptions,
             int insertionPosition,
             CancellationToken cancellationToken)
         {
             var bracePair = GetBracePairString(formattingOptions);
 
             // 1. Insert { }.
-            var newDocument = document.InsertText(insertionPosition, bracePair, cancellationToken);
```

Comment:
oh god.  these extensions are terrible.  tehy're actually mutating the workspace :(  we should kill all of htem.

Reply:
resolved.  you killed all of them :D